pkg/kubelet/volumes.go

```diff
@@ -167,6 +167,27 @@ type volumeTuple struct {
 	Name string
 }
 
+// ListVolumesForPod returns a slice of the volumes associated with the given pod
+func (kl *Kubelet) ListVolumesForPod(podUID types.UID) (map[string]volume.Volume, error) {
+	// Get the volume (kind, names) so we can instantiate Volume objects
+	tuples, err := kl.getPodVolumes(podUID)
+	if err != nil {
+		return nil, err
+	}
+	volumes := map[string]volume.Volume{}
+	for _, v := range tuples {
+		// VolumeCleaner implements the VolumeInterface so reuse this method
+		// TODO: Refactor this code so it isn't making this confusing call
+		// Instantiate a volume for each volume typle
+		cleaner, err := kl.newVolumeCleanerFromPlugins(v.Kind, v.Name, podUID)
```

Comment:
After more looking into this, the VolumeBuilder requires a lot of additional information since it is expected to build the volume.  Ideally the plugin interface would expose a newVolume() method.  For the moment I just created a wrapper method - kl.newVolumeFromPlugins to quarantine this issue from the rest of the PR.